### PR TITLE
Document the keeplink parameter of POST /source/<project>/<package>?cmd=commitfilelist

### PR DESCRIPTION
**Pull request by dcermak**

### CODE REVIEW

File: docs/api/api/api.txt [hunk 1]
```diff
@@ -889,6 +889,7 @@ POST /source/<project>/<package>?cmd=commitfilelist
 Parameters:
   withvalidate: activate sha validation code (optional, but recommended)
   comment: comment for history, optional
+  keeplink: bool, optional. If the package is a link and keeplink is true, then the link is preserved. By default keeplink is false and the link is forgotten on when committing a new revision.
```
Comment (adrianschroeter) [inline docs/api/api/api.txt:892]: I would not write "forgotten", but "the entire content is replaced what can replace or remove a _link file"

**Comment by adrianschroeter** [comment]
and while you are over it, please add also:

linkrev: optional, specifies link revision together with keeplink parameter.
repairlink: optional, for solving conflicts (obsolete for branches)
withvalidate: recommended, enforces sha256 source validation.

**Comment by adrianschroeter** [comment]
thanks ! :)

**Comment by marcus-h** [comment]
On 2020-10-13 23:32:53 -0700, Adrian Schröter wrote:
> repairlink: optional, for solving conflicts (obsolete for branches)

Hmm repairlink makes also sense for branches. For instance, it is needed
if the commit is going to solve a merge conflict (in such case, one
should also specify a linkrev).

**Comment by adrianschroeter** [comment]
On Mittwoch, 14. Oktober 2020, 09:23:51 CEST Marcus Hüwe wrote:
> On 2020-10-13 23:32:53 -0700, Adrian Schröter wrote:
> > repairlink: optional, for solving conflicts (obsolete for branches)
> 
> Hmm repairlink makes also sense for branches. For instance, it is needed
> if the commit is going to solve a merge conflict (in such case, one
> should also specify a linkrev).

right .. so let's phrase it that way

repairlink: optional, for solving conflicts. additional linkrev parameter is recommended.

-- 

Adrian Schroeter <adrian@suse.de>
Build Infrastructure Project Manager

SUSE Software Solutions Germany GmbH,  Maxfeldstr. 5, 90409 Nuernberg, Germany
(HRB 247165, AG München), Geschäftsführer: Felix Imendörffer

File: docs/api/api/api.txt (hunk 1 above)
Comment (dcermak) [reply docs/api/api/api.txt]: Fixed

**Review by vpereira — APPROVED** [review]

**Pull request merged**